homeassistant/components/iammeter/config_flow.py

```diff
@@ -0,0 +1,123 @@
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_TIMEOUT = 8
+
+
+@callback
+def iammeter_entries(hass: HomeAssistant):
+    """Return the hosts already configured."""
+    return {
+        entry.data[CONF_NAME] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class IammeterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Iammeter."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.host = None
+        self._errors = {}
+        self.discovered_conf = {}
+
+    def _host_in_configuration_exists(self, host) -> bool:
+        """Return True if host exists in configuration."""
+        if host in iammeter_entries(self.hass):
+            return True
+        return False
+
+    async def _test_connection(self, host, port):
+        try:
+            with async_timeout.timeout(PLATFORM_TIMEOUT):
+                await iammeter.real_time_api(host, port)
+                return True
+        except (IamMeterError, asyncio.TimeoutError):
+            _LOGGER.error("Device is not ready!")
+            self._errors[CONF_NAME] = "cannot_connect"
+        return False
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+        self._errors = {}
+        if user_input is not None:
+            # set some defaults in case we need to return to the form
+            name = user_input.get(CONF_NAME, DEFAULT_NAME)
+            port = user_input.get(CONF_PORT, DEFAULT_PORT)
+            host_entry = user_input.get(CONF_HOST)
+            url = urlparse(host_entry, "http")
+            netloc = url.netloc or url.path
+            path = url.path if url.netloc else ""
+            url = ParseResult("http", netloc, path, *url[3:])
+            host = netloc
+
+            if self._host_in_configuration_exists(name):
+                self._errors[CONF_NAME] = "already_configured"
+            else:
+                if await self._test_connection(host, port):
+                    return self.async_create_entry(
+                        title=name,
+                        data={CONF_NAME: name, CONF_HOST: host, CONF_PORT: port},
+                    )
+        else:
+            user_input = {}
+            user_input[CONF_NAME] = DEFAULT_NAME
+            user_input[CONF_PORT] = DEFAULT_PORT
+            user_input[CONF_HOST] = DEFAULT_HOST
+            if self.discovered_conf:
+                user_input.update(self.discovered_conf)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_NAME, default=user_input.get(CONF_NAME, DEFAULT_NAME)
```

Comment:
I don't think we need to have a name option. The user can already change name of entities since we set a unique id. Just set a default name.

Reply:
Thanks for your advices,I have changed my codes.
waiting for your review.
Thanks a lot~